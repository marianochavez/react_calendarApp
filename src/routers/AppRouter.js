import React from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { LoginScreen } from '../components/auth/LoginScreen'
import { RegisterScreen } from '../components/auth/RegisterScreen'
import { CalendarScreen } from '../components/calendar/CalendarScreen'

export const AppRouter = () => {
  return (
    <>
      <BrowserRouter>
        <Routes>
            <Route path="/*" element={<CalendarScreen />} />
            <Route path="/login" element={<LoginScreen />} />
            <Route path="/register" element={<RegisterScreen />} />
        </Routes>
      </BrowserRouter>
    </>
  )
}
