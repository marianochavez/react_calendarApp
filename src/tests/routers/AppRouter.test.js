import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { AppRouter } from '../../routers/AppRouter';


const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );


// store.dispatch = jest.fn();


describe('Pruebas en <AppRouter />', () => {

    test('debe de mostrar el espere...', () => {
        
        const initState = {
            auth: {
                checking: true
            }
        };

        const store = mockStore( initState );

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.spinner-container').exists() ).toBe(true);
        
    })
    

    test('debe de mostrar la ruta pÃºblica', () => {
        
        const initState = {
            auth: {
                checking: false,
                uid: null
            }
        };
        
        const store = mockStore( initState );

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h2').text().trim()).toBe('Login');
        
    })

    test('debe de mostrar la ruta privada', () => {
        
        const initState = {
            calendar: {
                events: []
            },
            ui: {
                modalOpen: false
            },
            auth: {
                checking: false,
                uid: '123',
                name: 'Juan Carlos'
            }
        };
        
        const store = mockStore( initState );

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.calendar-screen').exists() ).toBe(true);
        
    })
    
})
