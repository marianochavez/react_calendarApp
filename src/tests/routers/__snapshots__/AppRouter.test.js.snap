// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el espere... 1`] = `
<div>
  <div
    className="spinner-container"
  >
    <svg
      fill="none"
      viewBox="0 0 248 247"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        id="spinner2"
      >
        <g
          id="outer"
          style={
            Object {
              "animationDuration": "5s",
            }
          }
        >
          <circle
            cx="123.5"
            cy="124.5"
            id="Ellipse 1"
            r="104"
            stroke="#DABDFF"
            strokeWidth="11"
          />
          <circle
            cx="124"
            cy="20"
            fill="#8C37F7"
            id="Ellipse 2"
            r="20"
          />
          <circle
            cx="124"
            cy="227"
            fill="#8C37F7"
            id="Ellipse 5"
            r="20"
          />
          <circle
            cx="228"
            cy="127"
            fill="#8C37F7"
            id="Ellipse 3"
            r="20"
          />
          <circle
            cx="20"
            cy="127"
            fill="#8C37F6"
            id="Ellipse 4"
            r="20"
          />
        </g>
        <g
          id="inner"
          style={
            Object {
              "animationDuration": "5s",
            }
          }
        >
          <circle
            cx="122.916"
            cy="125.545"
            id="Ellipse 1_2"
            r="61.1714"
            stroke="#C4C4C4"
            strokeWidth="11"
            transform="rotate(32.5155 122.916 125.545)"
          />
          <circle
            cx="157.374"
            cy="72.0558"
            fill="#DABDFF"
            id="Ellipse 2_2"
            r="12.1774"
            transform="rotate(32.5155 157.374 72.0558)"
          />
          <circle
            cx="89.626"
            cy="178.335"
            fill="#DABDFF"
            id="Ellipse 5_2"
            r="12.1774"
            transform="rotate(32.5155 89.626 178.335)"
          />
          <circle
            cx="175.751"
            cy="161.03"
            fill="#DABDFF"
            id="Ellipse 3_2"
            r="12.1774"
            transform="rotate(32.5155 175.751 161.03)"
          />
          <circle
            cx="68.958"
            cy="92.9549"
            fill="#DABDFF"
            id="Ellipse 4_2"
            r="12.1774"
            transform="rotate(32.5155 68.958 92.9549)"
          />
        </g>
      </g>
    </svg>
    <p>
      Loading
    </p>
  </div>
</div>
`;

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta privada 1`] = `
<div
  className="calendar-screen"
>
  <nav
    className="navbar navbar-dark bg-dark navbar-expand-md justify-content-center"
  >
    <div
      className="container"
    >
      <p
        className="navbar-brand d-flex w-50 m-auto"
      >
        Bienvenido, 
        Juan Carlos
        !
      </p>
      <button
        className="navbar-toggler"
        data-bs-target="#collapsingNavbar"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="navbar-collapse collapse w-100"
        id="collapsingNavbar"
      >
        <ul
          className="nav navbar-nav ms-auto w-100 justify-content-end"
        >
          <li
            className="nav-item ms-auto"
          >
            <button
              className="nav-link text-danger btn "
              onClick={[Function]}
            >
              Logout  
              <i
                className="fa-solid fa-arrow-right-from-bracket"
              />
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Hoy
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &lt;
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &gt;
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        marzo 2022
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className="rbc-active"
          key="month"
          onClick={[Function]}
          type="button"
        >
          Mes
        </button>
        <button
          className=""
          key="week"
          onClick={[Function]}
          type="button"
        >
          Semana
        </button>
        <button
          className=""
          key="day"
          onClick={[Function]}
          type="button"
        >
          Día
        </button>
        <button
          className=""
          key="agenda"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      aria-label="Month View"
      className="rbc-month-view"
      role="table"
    >
      <div
        className="rbc-row rbc-month-header"
        role="row"
      >
        <div
          className="rbc-header"
          key="header_0"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            lun.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_1"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mar.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_2"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mié.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_3"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            jue.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_4"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            vie.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_5"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            sáb.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_6"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            dom.
          </span>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                28
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                04
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                05
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                06
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                07
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                08
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                09
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                10
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                11
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                12
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                13
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                14
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                15
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                16
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                17
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                18
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                19
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                20
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                21
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                22
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                23
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                24
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                25
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                26
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                27
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-today"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                28
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                29
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                30
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-now rbc-current"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                31
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="btn btn-primary fab"
    onClick={[Function]}
  >
    <i
      className="fas fa-plus"
    />
  </button>
  <Portal
    containerInfo={
      <div
        class="ReactModalPortal"
      />
    }
  />
</div>
`;

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta pública 1`] = `
<section
  className="vh-100 gradient-custom"
>
  <div
    className="container py-5 h-100"
  >
    <div
      className="row d-flex justify-content-center align-items-center h-100"
    >
      <div
        className="col-12 col-md-8 col-lg-6 col-xl-5"
      >
        <div
          className="card card-login bg-dark text-white animate__animated animate__fadeIn animate__fast"
        >
          <div
            className="card-body p-5 text-center"
          >
            <div
              className="mb-md-5 mt-md-4 pb-5"
            >
              <h2
                className="fw-bold mb-2 text-uppercase"
              >
                Login
              </h2>
              <p
                className="text-white-50 mb-5"
              >
                Ingresar email y contraseña!
              </p>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-outline form-white mb-4"
                >
                  <input
                    autoComplete="off"
                    className="form-control form-control-lg"
                    id="typeEmailX"
                    name="lEmail"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value="mariano@gmail.com"
                  />
                </div>
                <div
                  className="form-outline form-white mb-4"
                >
                  <input
                    autoComplete="off"
                    className="form-control form-control-lg"
                    id="typePasswordX"
                    name="lPassword"
                    onChange={[Function]}
                    placeholder="Contraseña"
                    type="password"
                    value="123456"
                  />
                </div>
                <button
                  className="btn btn-outline-light btn-lg px-5"
                  type="submit"
                >
                  Ingresar
                </button>
              </form>
            </div>
            <div>
              <p
                className="mb-0"
              >
                No tienes cuenta?
                 
                <a
                  className="text-white-50 fw-bold"
                  href="/register"
                  onClick={[Function]}
                >
                  Registrarme
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
